### PR TITLE
Some fixes and improvements for better usage

### PR DESCRIPTION
**Pull request by Jeehut**
Thank you for this project, it was very useful for me implementing the automatic translation functionality on my open source project [BartyCrouch](https://github.com/Flinesoft/BartyCrouch).

When using this library I came across a few minor issues though which I all managed to fix. This pull request contains all those fixes – they are all minor but important changes. Please have a look at the commits for further information but here's a short overview of the commit messages first lines:
- Stop overriding explicit fromLanguage with fallback
- Move thread choosing logic to callback closure
- Add support for Bosnian and Serbian languages
- Add Language creator method from locale

You can see how I use Polyglot in the `translateEmptyValues` method within [this class](https://github.com/Flinesoft/BartyCrouch/blob/stable/Sources/Code/StringsFileUpdater.swift).

If you have any questions feel free to ask. I hope this helps.

### CODE REVIEW

**Comment by ayanonagon** [comment]
Thanks for the PR! It looks like like one of the tests started failing. Mind taking a look? :)

**Comment by Dschee** [comment]
It fails because of my change of API: The translate callback is no longer called on the main thread – see my commit message on a078ea9. I think you should review if this change is okay or not. If it is okay it is easy to change the tests accordingly, I can do that then. What do you think?

**Comment by ayanonagon** [comment]
1. Backwards compatibility
Since this project hasn’t reached 1.0 yet, I’m OK with making small(ish) breaking changes. I can see why not everyone wants to receive the callback on the main queue. I will note this change in the release notes.

2. `forLocale`
Can we rename this to `languageForLocale` and rename the first parameter to `languageCode`? Also tests for this would be :sparkles:

Thanks for adding Bosnian and Serbian as well!

**Comment by Dschee** [comment]
Thanks for reviewing. I plan do the suggested changes and update the pull request within the next two hours. Stay tuned ... :)

**Comment by Dschee** [comment]
I have done the changes you requested and even added a few more languages (like Croatian or Kiswahili). Although I also changed the test to call the main thread in the block it didn't pass on travis. It is passing on my machine locally though so I can't debug the issue. Do you have any ideas what this might be?

**Comment by ayanonagon** [comment]
That’s really weird! What do you think about moving the main queue change in to a separate PR so we can diagnose it separately and get you other changes merged in first? Thanks for adding all the tests too. :D

**Comment by Dschee** [comment]
Yeah, let's do this like you suggested.

File: Polyglot/Polyglot.swift [hunk 1]
```diff
@@ -128,7 +172,7 @@ public class Polyglot {
                 defer {
                     dispatch_async(dispatch_get_main_queue()) {
                         callback(translation: translation)
-                    }
+                    }
```
Comment (ayanonagon) [inline Polyglot/Polyglot.swift]: Can you get rid of the trailing white space here?

File: Polyglot/Polyglot.swift [hunk 2]
```diff
@@ -97,14 +138,17 @@ public class Polyglot {
     }
 
     /**
-        Translates a given piece of text.
+        Translates a given piece of text asynchronously. Switch to the main thread within the callback
+        if you want to update your UI by using `dispatch_async(dispatch_get_main_queue()) { /* code */ }`.
```
Comment (ayanonagon) [inline Polyglot/Polyglot.swift]: Can you revert the part about the main queue from the docs as well? :D

**Comment by ayanonagon** [comment]
Two small comments, and I’ll merge! Thanks for making the changes. :rocket:

**Comment by Dschee** [comment]
Done – ready for merge now, I hope. :checkered_flag:

**Comment by ayanonagon** [comment]
:green_apple: :rocket: 

Thanks again for this PR!! Awesome stuff. :smile:

**Pull request merged**